pyaptly/state_reader.py

```diff
@@ -77,65 +60,64 @@ def read_gpg(self):
             if field[0] in ("pub", "sub"):
                 key = field[4]
                 key_short = key[8:]
-                self.gpg_keys.add(key)
-                self.gpg_keys.add(key_short)
+                gpg_keys.add(key)
+                gpg_keys.add(key_short)
+        return gpg_keys
 
-    def read_publish_map(self):
-        """Create a publish map. publish -> snapshots."""
-        self.publish_map = {}
-        # match example:  main: test-snapshot [snapshot]
+    @lru_cache(None)
```

Comment:
Use named parameter to better document what's going on

```suggestion
    @lru_cache(maxsize=None)
```